### DISCUSSION TITLE
Is there a way to create module? (group of APIs)

### DISCUSSION THREAD

**Discussion by darron1217** [strapi/strapi#7799] [Legacy Topics]
I tried local plugin to acheive it, but core-api does not work on plugin.

Is there a way to make `strapi generate:api Test --plugin my-plugin` work properly?
(I need automatic api&graphql genration like `strapi generate:api Test`)

**Comment by derrickmehaffy** [comment]
I'm not sure I understand what you are trying to achieve, can you explain in more detail?

**Reply by darron1217** [reply]
The command `strapi generate:api Test` generates REST API(routes) and graphql magically. (This is awesome)

But I want to move the folders under `./api` to plugin.

Final goal is to make plugin like `ecommerce`(all APIs are ready-to-use) so that I can just install it and use it.

**Comment by derrickmehaffy** [comment]
The plugins don't have access to the `core-api` which provides all the default routes, controllers, and services. You can do what you have stated and generate the model normally, move it to the plugin, but you will need to copy the core controllers and services to that model.

https://strapi.io/documentation/v3.x/concepts/services.html#concept
https://strapi.io/documentation/v3.x/concepts/controllers.html

All of the core ones are listed in those two documentation pages.

**Reply by darron1217** [reply]
Thanks for your fast response 👍 

It's a bit unfortunate, but I hope it will be updated next time.
